src/render/index.js

```diff
@@ -116,6 +120,22 @@ class App extends React.PureComponent {
     });
   }
 
+  handleDatasetLoading(ast) {
+    const dataNodes = ast.children.filter(child => child.type === 'data');
+    if (dataNodes && dataNodes.length > 0) {
+      dataNodes.forEach(node => {
+        const { name, source } = node.properties;
+        const { content } = readFile(source.value);
+
+        if (content) {
+          this.getContext().context.update({
+            [name.value]: jsonParser(content)
+          });
```

Comment:
Not essential, but it may be more performant to read all the files in the loop and then send single `context.update({...})` call after

Reply:
Ahh good point! I'll update this